dvc/repo/experiments/executor/base.py

```diff
@@ -464,6 +466,23 @@ def checkpoint_callback(
             exp_rev = cls.commit(
                 scm, exp_hash, exp_name=name, force=force, checkpoint=True
             )
+
+            git_remote = os.environ.get("DVC_EXP_AUTO_PUSH", None)
+            if git_remote:
+                from dvc.repo.experiments.push import push
+
+                branch = dvc.experiments.get_branch_by_rev(
+                    exp_rev, allow_multiple=None
+                )
+                branch_name = ExpRefInfo.from_ref(branch).name
+                push(
+                    dvc,
+                    git_remote,
+                    branch_name,
+                    push_cache=True,
+                    run_cache=True,
+                )
+                logger.info({"pushed": branch_name})
```

Comment:
This push also needs to happen for the final `executor.commit()` at the end of `executor.reproduce()` (line 359).

So maybe this env var check + push should maybe be done inside `commit()` so we don't end up missing additional calls in the future in case we add more places that commits can happen.

Reply:
This would also create more consistency between checkpoint and non-checkpoint experiments in cml. `dvc exp push` probably shouldn't be needed in the workflow in either case.